### ISSUE TITLE
[UX] Rename name "Filtered HTML" input format to "Basic"

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#1188]
Let's remove the word "HTML" from the Filtered HTML format name.  Since you can't enter HTML here anymore, having a format name that indicates that you can is confusing.

I'm proposing we change the name of this format to "Basic".

Related issue: [UX] Change the name of the "Full HTML" text format to "Raw HTML" - https://github.com/backdrop/backdrop-issues/issues/4499 - 
EDIT [This related issue is done and merged into core]

**Comment by quicksketch** [comment]
If we do this, I'd prefer we did it right and renamed the underlying format machine name as well, which will break a lot of tests (I'm guessing here). While it would be great to get this in before *tomorrow*, time is short and this could easily wait until 1.3.0.

**Comment by jenlampton** [comment]
Yea, I looked at all the things we'd need to change to change the name, and I don't think it'll happen tonight.

**Comment by jenlampton** [comment]
I would like to make this change in the Standard install profile, but not for people who have existing sites as that might be too confusing. Pushing an updated PR that changes both the label and the underlying machine name of the format.

**Comment by klonos** [comment]
I'm all for this 👍 ...the PR sandbox looks good, but there are still test failures.

**Comment by jenlampton** [comment]
Rebased, and pushed a clean PR.

**Comment by stpaultim** [comment]
Sandbox needs refresh!

**Comment by indigoxela** [comment]
It's not just a sandbox refresh, there are conflicts and possibly even failing tests. Back to "needs work".

@stpaultim and @jenlampton I'm aware you're currently looking for "good first issues", but I wouldn't ever *ever* call anything regarding fixing tests easy. :wink:

**Comment by indigoxela** [comment]
Apart of that a feature request shouldn't get the label "milestone candidate bug".

**Comment by stpaultim** [comment]
We just changed [Full HTML to Raw HTML](https://github.com/backdrop/backdrop-issues/issues/4499) 
(PR committed this week). Does that increase or decrease the need for this change? I think the other changes was more urgent than this one, but I think I would be in support of this one as well.

**Comment by klonos** [comment]
From the issue summary:

> Let's remove the word "HTML" from the Filtered HTML format name. Since you can't enter HTML here anymore, having a format name that indicates that you can is confusing.

Seeing that that was written back in 2015, is that still the case? Looking at `admin/config/content/formats/filtered_html` I see that the "Source code" button is added by default to the CKEditor toolbar, and the "Limit allowed HTML tags" filter is added to the format (with the following allowed: `<a> <em> <strong> <cite> <blockquote> <code> <ul> <ol> <li> <dl> <dt> <dd> <h3> <h4> <h5> <p> <img> <figure> <figcaption>`). This to me means the ability to add HTML 🤷

**Comment by jenlampton** [comment]
I still think the argument is valid. What they put in the textarea gets
filtered, regardless of whether they type any HTML or not.

Most people who use these forms don't know HTML. We should choose a label
that resonates with them too, and not just the web savvy.

**Comment by BWPanda** [comment]
I'm OK with changing this. My suggestion would to be to do what we did what we did with Full HTML, i.e. change the human-readable name now, then change the machine name at the same time we change the Full HTML machine name.

**Comment by klonos** [comment]
Fair enough. It's just that I feel that "Basic" may be interpreted as "Plain text". Would "Rich text" be better? ...although I don't really like using the term "rich" 🤔 ...what do regular, non-techie forks (content editors) use for what we call a WYSIWYG?

**Comment by philsward** [comment]
Maybe a blessing in disguise is to rename the human name before the machine name, that way we can add a human and if it gets feedback there's time to change it before the machine name gets changed.

I came up with a few names I was going to share, but they ultimately all led back to synonyms of Basic, so I'm good with sticking to basic.

**Comment by indigoxela** [comment]
Yet another issue about renaming something, I simply don't understand...

Just like in the other issues: keep in mind that we can *not* change the machine name, for sure not in a minor release. @jenlampton I'd suggest that you revert that in your PR.

**Comment by stpaultim** [comment]
> Yet another issue about renaming something, I simply don't understand...

I understand the frustration and would be happy to try to explain my (subjective) position on this, but I suspect that you are not really looking for that right now. ;-)

I agree that there are costs and benefits to renaming things and hope that we are able to some to good renaming decisions based upon balancing the costs vs benefits on a case by case basis. 

> Just like in the other issues: keep in mind that we can not change the machine name, for sure not in a minor release. @jenlampton I'd suggest that you revert that in your PR.

In our extensive discussions about this for the [full_html to raw_html](https://github.com/backdrop/backdrop-issues/issues/4499) issue, I think we've decided that in THEORY we should be able to change the machine names for text formats, because neither contrib or custom developers should be hard coding text formats into their modules. Text formats are nothing more than default config created by the standard install profile and can be changed on any site. Developers should never assume the existence of a specific text format (or machine name for a text format).

Having said that. We know that in the case of full_html, some contrib modules have hard coded assumptions about text formats and we have decided to be fairly cautious about making this kind of change. We do have an open issue about how to change the machine name of the full_html format.  

https://github.com/backdrop/backdrop-issues/issues/4806

It is my understanding, that we would consider changing the machine name of default text formats prior to 2.0, but only after someone spent time doing the legwork to see what kind of impact that this would have on these contrib modules. We have a list of the current contrib modules that reference the "full_html" machine name in https://github.com/backdrop/backdrop-issues/issues/4806.

**Comment by philsward** [comment]
@indigoxela I can't speak for others, but I'm all in with the main idea behind these subtle changes.

Backdrop should be easier to understand for the people who "use" it. I'm not referring to the builders, themers, coders or even moderators, but rather the end-user who uses the platform after someone else has done all the work.

These changes aren't for you or I or anyone here but rather the people who don't know anything about the web and don't care, but still have to use it.

Usually by the time a developer gets done working on all the code to make something work, UX and general communication are the last priority, so something gets stuck in there and called "good enough" until someone else comes along and says: "It's time for it to get fixed." Which is where we are now.

**Comment by indigoxela** [comment]
An example: if you export a single view, and it has some (help) text in header or footer, or a filtered text for "no results", the format is included in the json file. So exported views would break, if the format isn't there on the site where the view is imported (unless the test format definition also gets imported). This could be problematic if a site builder re-uses the same view on a newer site and the site they created the view has the format (it shipped with core) and the newer site does not have the format (got renamed). The same applies to views that ship with a module - for contrib, but also for custom ones. How would you catch the latter?

Another exampe: I'm pretty sure, many functional tests out there rely on filtered_html format. "Value the needs of the editor and architect over the needs of the developer" does - to my understanding - not mean "discard the needs of the developer". And changes like this would exactly do that.

Anyway, go for it.

**Comment by findlabnet** [comment]
>These changes aren't for you or I or anyone here but rather the people who don't know anything about the web and don't care, but still have to use it.

Just curious to know who are these people?

**Comment by stpaultim** [comment]
@indigoxela - My comments above were meant to summarize my understanding of what was jointly decided after discussion in at least 2 (probably more) dev meetings and additional discussion in the issue queue. I was lead to believe by the group (including several core developers), that changing the machine name short of 2.0 was possible (and preferable), but ONLY if we met some fairly specific criteria. (There is an issue making this explicit https://github.com/backdrop/backdrop-issues/issues/4806 and an opportunity for anyone to give feedback.)

The ONLY reason we were/are considering changing the machine name is to reduce the confusion and difficulties for developers. Changing the machine does not help editors and architects. 

> "Value the needs of the editor and architect over the needs of the developer" does - to my understanding - not mean "discard the needs of the developer". And changes like this would exactly do that.

I resent the implication that somehow suggesting that we might be able to change the machine name of text formats is a demonstration of ours (or me) discarding the needs of developers. 

If as a developer, you believe that changing the machine name is a bad idea under any conditions, I'm fine with that. I want to hear those reasons and take them into account. But, I don't recall hearing that from you in the past and I don't recall anyone putting forward these specific reasons for not doing it. I look forward to hearing what others think. I am happy to change my opinion based upon additional input. My only goal in even suggesting that we change the machine name is to balance the needs of editors/architects and developers, not to dismiss the needs of anyone.

**Comment by indigoxela** [comment]
> If as a developer, you believe that changing the machine name is a bad idea under any conditions, I'm fine with that. I want to hear those reasons and take them into account. But, I don't recall hearing that from you in the past

To clarify: I'm not only developer, but also site builder and user.

I wasn't involved much in the issue regarding renaming "full_html", but I [did raise a concern that it might be a contrib breaker](https://github.com/backdrop/backdrop-issues/issues/4499#issuecomment-699862816). The fact that I didn't object shouldn't get interpreted as consent.

That other PR _did_ not include the machine name change, but this one here does. That's why I lay more emphasis on my warnings.

I posted my concerns here in my previous comment. The problems I describe do not only affect developers, but also site architects.

**Comment by stpaultim** [comment]
> I wasn't involved much in the issue regarding renaming "full_html", but I did raise a concern that it might be a contrib breaker. The fact that I didn't object shouldn't get interpreted as consent.

AND, the discussion around final implementation of that PR was heavily focused on the impacts it might have on contrib. The final decision to NOT change machine names was totally based upon that concern. And the decision that we MIGHT in the future change the machine name was heavily tied to first understanding exactly how contrib might be effected and taking steps to mitigate it. Again, I was keeping that option open, because I thought it in the best interest of developers.

Yet, you are suggesting that the needs of developers are being discarded.

> That other PR did not include the machine name change, but this one here does. That's why I lay more emphasis on my warnings.

This PR was written over a year before any of the discussion in the other issue. What reason do you have to believe that after spending so much time discussing and responding to the possible implications to contrib, that we would then disregard these concerns for this issue. 

I am not at all objecting to your concerns. I welcome your technical concerns. I am objecting to the implication that that this discussion is somehow evidence that the needs of developers are being discarded.

In my opinion, I revisited this issue SIMPLY to gauge whether there is still any interest in this issue. We haven't even gotten close to updating the PR yet and I can't see why anyone would assume that this PR would not undergo the same scrutiny that the previous one did. I'm heavily puzzled at why this issue is being used to suggest that developers are being ignored.

Clearly, there must be a larger issue going on here.

**Comment by philsward** [comment]
> Just curious to know who are these people?

@findlabnet well, from my experience, 60yr olds who want me to build them website and don't know anything about HTML.

**Comment by indigoxela** [comment]
> The only real danger that I am aware of is to a new site that tries to use an existing contrib module with hard-coded references to text formats

@stpaultim  Like views that ship with a module. Views containing any fields with a filtered text format contain the machine name in their json files. How could module maintainers ever catch that? Especially if they have to consider several core versions, where some ship with the old machine name and others with the new one.

Again: changing the machine name prior to 2.x will - to my understanding - be impossible.

> 60yr olds who want me to build them a website, want to edit content on their own and don't know anything about "HTML".

@philsward Then change the label of that format. That's a 5-second-job. (It's a little discriminatory to assume that people beyond 60 don't know anything about HTML, BTW.)

**Comment by BWPanda** [comment]
Since @jenlampton's PR is causing some contention because it hasn't been updated to *not* change the machine name, I've made a separate PR that just changes the label and leaves the machine name as-is: https://github.com/backdrop/backdrop/pull/4096

**Comment by bugfolder** [comment]
Tested https://github.com/backdrop/backdrop/pull/4096. Applied the patch, installed a new Backdrop site. Went to edit one of the sample nodes:

![image](https://github.com/backdrop/backdrop-issues/assets/29214084/add88fb2-f61a-48fb-aafe-83bb38d54b12)

WFM. Code reviewed, LGTM.

I will close the other PR https://github.com/backdrop/backdrop/pull/2885, and will suggest that further discussion of changing the machine name (if needed) be in a separate issue.

**Comment by indigoxela** [comment]
I do have concerns, as the active config for that filter on existing sites isn't updated (needs update hook). Is the label supposed to get updated?

**Comment by bugfolder** [comment]
@jenlampton had said:

> I would like to make this change in the Standard install profile, but not for people who have existing sites as that might be too confusing.

So no update hook or change on existing sites.

**Comment by stpaultim** [comment]
@quicksketch asked for input on this issue today in the dev meeting. 

I'm in favor of this change, it's a follow-up to https://github.com/backdrop/backdrop-issues/issues/4499, when we changed Full HTML to Raw HTML. I don't think "Filtered HTML" means much to most site editors for what is in fact the "Basic" default text editor in Backdrop. 

We have the experience of changing "Full HTML" to "Raw HTML" to mimic. That change went pretty seamlessly, in my opinion. 

This will ONLY effect new sites as it's part of the standard install profile. Have we looked at the Raw HTML PR to see if there is anything we are missing from this one?

**Comment by stpaultim** [comment]
We did have this line in the previous PR to help document the change. It might be good to include the same here:

```
 // Changed display name from Full HTML to Raw HTML but keeping machine name
  // for backward compatibility.
```
I'm a little surprised that this PR is so much shorter than the previous one:
https://github.com/backdrop/backdrop/pull/3365/files

**Comment by klonos** [comment]
@stpaultim I've filed a fresh PR based off of the one Peter started. I rebased it, fixed PHPCS complaints (mostly lines exceeding 80 characters), and a few grammatical errors here and there in the relevant files/comments.

This commit is the rebase (no visual/actual changes, as there were no conflicts): https://github.com/backdrop/backdrop/pull/4576/commits/cda3d56a4731a532382ebac965f9d02ae3d79c67

This commit has the additional changes to what you previously tested/reviewed: https://github.com/backdrop/backdrop/pull/4576/commits/730208c90d5a35618831ecbac7f93d5be6cb9a02

> We did have this line in the previous PR to help document the change. ...

Yup, I've added a similar line in this PR as well. Thanks for pointing that out 👍🏼

**Comment by quicksketch** [comment]
Although I'm not totally sure about this, I think @stpaultim makes a good point that we've already done this once before. So I've merged https://github.com/backdrop/backdrop/pull/4576 into 1.x for 1.27.0.

I retitled this issue to indicate we only changed the label, not the underlying machine  name.

https://github.com/backdrop/backdrop/commit/5fd0697bd210b8934363af3dd710366277996347 by @BWPanda, @klonos, @jenlampton, @indigoxela, @stpaultim, @philsward, and @bugfolder.

Thanks folks!
